### PR TITLE
@miketheman configurable process monitoring

### PR DESCRIPTION
**Pull request by jpcallanta**
I've added process.rb recipe as well as the process.yaml.erb as discussed in the thread for pull request #124 and added test-kitchen test.

### CODE REVIEW

File: recipes/process.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#   },
+#   {
+#     :name => "postgres",
+#     :search_string => ["postgres"],
+#   },
+# ]
+
+datadog_monitor 'process' do
+  instances node['datadog']['process']['instances']
+end
```
Comment (miketheman) [inline recipes/process.rb]: rubocop might yell at you for lacking a newline here

File: .kitchen.yml [hunk 2]
```diff
@@ -222,3 +222,21 @@ suites:
...
+  run_list:
+    - recipe[datadog::process]
+  attributes:
+    datadog:
+      <<: *DATADOG
+      process:
+        instances:
+          - name: "pidname"
+            exact_match: false
+            git a
```
Comment (miketheman) [inline .kitchen.yml]: this might not be intended

**Comment by miketheman** [comment]
Nice stuff. I might add a link in the "docs" section to see more over here: https://github.com/DataDog/dd-agent/blob/master/conf.d/process.yaml.example

File: recipes/process.rb (hunk 1 above)
Comment (jpcallanta) [inline recipes/process.rb]: Ah, I'll fix

**Comment by miketheman** [comment]
Thanks! I rebased and cleaned up a bit for brevity, and merged in f048bbb62b16f69cfa1fca06228b3a93fc7152f4

**Pull request closed without merging**